### ISSUE TITLE
Add a `minimizeToTray` action

### ISSUE THREAD

**Issue by rashil2000** [microsoft/terminal#11229]
# Description of the new feature/enhancement

When the `minimizeToTray` property is set to true, the window gets minimized to the tray on clicking the minimize button on top right. I feel this is not always a desirable response - people might want to minimize to taskbar (the normal behavior) or minimize to tray, separately. An action called `minimizeToTray` (and a way to set a keybinding for this) would work best - allowing for both behaviors according to what the users want.
This will also allow WT to behave similar to tmux's `detach` - where detaching and minimizing (to taskbar) are distinct behaviors.

# Proposed technical implementation details (optional)

I was reading the discussion here at https://github.com/microsoft/terminal/pull/10368#issuecomment-884294259 - was this action not implemented?

**Comment by rashil2000** [comment]
I've been meaning to make a contribution to WT for a long time - so I wanted to ask - if this isn't already implemented/being worked on, would it make a good beginner's issue?

**Comment by zadjii-msft** [comment]
Part of the reason why that action wasn't implemented during that PR is because we weren't totally sold on how it should all work. The only way it could work right now is if the user already has `"alwaysShowTrayIcon": true`. If that seems reasonable to you, then I don't think it'd be terribly difficult to implement. 

However, trying to implement this action to work even with `"alwaysShowTrayIcon": false` would be a good amount harder. We'd need to track per-window if they wanted to be minimized to the ~~tray~~ Notification Area, and decide to display the ~~tray~~ Notification Area icon if at least one of them wanted to be in the ~~tray~~ Notification Area.

Also, the action should technically be named `minimizeToNotificationArea`, see #11219

**Comment by rashil2000** [comment]
> Part of the reason why that action wasn't implemented during that PR is because we weren't totally sold on how it should all work. The only way it could work right now is if the user already has `"alwaysShowTrayIcon": true`. If that seems reasonable to you, then I don't think it'd be terribly difficult to implement.

What would be the response if a user tries to invoke the `minimizeToNotificationArea` action when this setting is false?

**Comment by zadjii-msft** [comment]
I can think of two things:
* nothing at all
* the terminal is minimized (but just like, normally minimized, to the taskbar)

**Comment by rashil2000** [comment]
I feel doing _nothing_ is better than doing _something else_.


On an unrelated note: pardon me for the stupid question, but what exactly happens when an app goes to the tray? Since the window is no longer visible, does it mean it is consuming less resources (which would have been used otherwise for drawing/painting the window and/or managing related memory/CPU time)? Of course the pwsh (or equivalent) process will continue to consume the same resource as before as it's still running in the background.

**Comment by zadjii-msft** [comment]
>  but what exactly happens when an app goes to the tray

I'm pretty sure that nothing special happens. It's basically just the same as being in the taskbar, but with one less window snapshot to draw in alt+tab & win+tab. So I'd presume the difference is negligible.

**Comment by leonMSFT** [comment]
Yeah I originally had the action implemented but left it out of the initial PR to stew on it a bit more. Then I got caught up in doing other things and forgot to put out a separate PR with this action 😅

IMO I like the idea of having the action work even with `alwaysShowNotificationIcon:false`, because if you're explicitly trying to `minimizeToNotificationArea` with an action, you know what you want, and there shouldn't be a global setting that you might not be aware of that's blocking your action. It should just do the thing and pop in a notification icon and the icon should just disappear when it's no longer in the notification area.

Though with that behavior, like @zadjii-msft said, it would require per-window tracking of minimized or not.

**Comment by rashil2000** [comment]
> ...and there shouldn't be a global setting that you might not be aware of that's blocking your action.

I was thinking about this too - because currently IIRC we don't have any actions that are dependent on a global setting.

> Though with that behavior, like @zadjii-msft said, it would require per-window tracking of minimized or not, making it more complicated to impl. If it's something you're interested in taking on though, by all means go for it!

I'll be honest - I think it'll be too much for me 😅. I just know college-level C++ and have never worked on any project this large/complex. 

I have one other (unrelated) feature request that might be easier to get me started (I'll open an issue sometime later today).

**Comment by zadjii-msft** [comment]
Discussed with the team:
* We think this is a fine idea
* Ideally, performing this action would minimize to the tray regardless. If the tray icon isn't already visible (i.e. `alwaysShowNotificationIcon` is false), then it should show the icon.
* We'd probably be willing to accept just the "just minimize if there isn't a tray icon yet" and work towards the ideal implementation incrementally
* a90d2ec377e4c0af5227ac2dc860ccdc79fc686c is probably the best place to start. You'd probably want to revert that commit, though, there's been a fair amount of churn in that area, so it might not be that simple.
